src/error-table.rkt

```diff
@@ -1,20 +1,460 @@
+                 (infinite? y)
+                 (not (nan? subexpr-val)))
+            (mark-erroneous! subexpr)]
+
+           ;; If one operand is normal and the other overflows then, inf rescue
+           ;; could occur
+           [(and (or (infinite? x)
+                     (infinite? y))
+                 (not (infinite? subexpr-val)))
+            (mark-erroneous! subexpr)]
+
+           [(and (or (zero? x)
+                     (zero? y))
+                 (not (zero? subexpr-val)))
+            (mark-erroneous! subexpr)]
+
+           ;; If both normal then no error
+           [else #f])]
+        
+        [(list (or 'log.f64 'log.f32) arg)
+         #:when (list? arg)
+         (define arg-val (exacts-ref arg))
+         (define cond-num (bfabs (bf/ 1.bf
+                                      (bf subexpr-val))))
+         (cond
+           ; Condition number hallucination:
+           ; Condition number is high when x = 1,
+           ; but x is exactly 1, so there is no error
+           [(and (= arg-val 1.0) (= subexpr-val 0.0)) #f]
+           
+           ; overflow rescue:
+           [(infinite? arg-val) (mark-erroneous! subexpr)]
+           
+           ; underflow rescue:
+           [(zero? arg-val) (mark-erroneous! subexpr)]
+
+           ; High Condition Number:
+           ; CN(log, x) = |1 / log(x)|
+           [(bf> cond-num cond-thres)
+            (mark-erroneous! subexpr)]
+           
+           [else #f])]
+        
+        [(list (or 'exp.f64 'exp.f32) arg)
+         #:when (list? arg)
+         (define arg-val (exacts-ref arg))
+         
+         (cond
+           ; Condition Number Hallucination:
+           ; When x is large enough that exp(x) overflows,
+           ; condition number is also high.
+           [(and (infinite? (exp arg-val))
+                 (infinite? subexpr-val))
+            #f]
+
+           ; Condition Number Hallucination:
+           ; When x is large enough (negative) that exp(x)
+           ; underflows, condition number is also high
+           [(and (= (exp arg-val) 0.0)
+                 (= subexpr-val 0.0))
+            #f]
+
+           ; High Condition Number:
+           ; CN(exp, x) = |x|
+           [(> (abs arg-val) 100)
+            (mark-erroneous! subexpr)]
+           
+           [else #f])]
+
+        ; FIXME need to rework from scratch
+        [(list (or 'pow.f64 'pow.f32) x-ex y-ex)
+         #:when (or (list? x-ex) (list? y-ex))
+         (define x (exacts-ref x-ex))
+         (define y (exacts-ref y-ex))
+         (define x.bf (bf x))
+         (define y.bf (bf y))
+         (define x^y (expt x y))
+         (define cond-x (bfabs y.bf))
+         (define cond-y (bfabs (bf* y.bf (bflog x.bf))))
+
+         (cond
+           ;; Hallucination:
+           ;; x has a large exponent and y is 1. The ylogx is large but there is
+           ;; no error because the answer is exactly x
+           [(and (= y 1.0)
+                 (= x subexpr-val)) #f]
```

Comment:
This should really be some kind of exactness reasoning, right? Instead of `(= y 1.0)` it should be `(exact-expr? y)` or whatever? (Consider the case of `(pow x (+ 1.0 1e-20))`, where the `y` parameter rounds to 1.0 but is not exact.

Reply:
Yup it needs to be exactly 1, just like the log case.